test/framework/e2e_framework.go

```diff
@@ -47,61 +78,52 @@ type E2eFramework struct {
 }
 
 func NewE2eFramework(t *testing.T) (*E2eFramework, error) {
-	configFile, err := filepath.Abs("../../build/kubeconfig")
+	baseDir, err := filepath.Abs(filepath.Join("..", ".."))
 	if err != nil {
 		return nil, err
 	}
+	configFile := filepath.Join(baseDir, *configFile)
 
 	if _, err = os.Stat(configFile); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("cannot stat %s: %w", configFile, err)
 	}
 
 	config, err := clientcmd.LoadFromFile(configFile)
 	if err != nil {
 		return nil, err
 	}
 
-	controlPlaneContext := ""
 	var controlPlaneClient client.Client
 	remoteClients := make(map[string]client.Client, 0)
 	t.Logf("Using config file: %s", configFile)
 
-	for name, _ := range config.Contexts {
+	for name := range config.Contexts {
```

Comment:
I'm sticking my nose in here because I wanted to warn @adutra about the issues we've previously seen where tests end up getting run on the wrong cluster. Obviously this makes things quite dangerous for users who run production clusters, and people like myself who sometimes have temporary access to client clusters.

I agree with John that the control and data planes should have to be explicitly set, but I also think that it is safer to always use `build/kubeconfig` and make sure that no `kubectl` commands ever run against contexts from `~/.kube/config`. If a small bash script is required to copy named contexts from `~/.kube/config` into `build/kubeconfig` for tests run on remote clusters, then that might bridge the difference?

Reply:
I am fine with allowing a configurable kubeconfig file. We could require it to always be `build/kubeconfig`, but then if I want to be lazy, I might just copy `~/.kube/config`. Tools read from and write to the default kubeconfig file. On that basis I think it is reasonable to be able to use it provided we are being explicit about which clusters we are using.

As for a script to copy contexts, there are other elements that need would need to be copied. Yes it is doable, but it is additional maintenance. IMO, that extra maintenance isn't worth it as long as we are being explicit in our tests.

Reply:
I am going to add an allow-list of context names to create clients for.